source/code/plugins/oms_common.rb

```diff
@@ -478,6 +482,118 @@ class Common
     @@tzRightFolder = 'right/'
 
     class << self
+
+      # Internal methods
+      # (left public for easy testing, though protected may be better later)
+
+      def clean_hostname_string(hnBuffer)
+        hostname_buffer = hnBuffer.strip
+        return hostname_buffer
+      end
+
+      def has_designated_hostnamefile?
+        return false if @@HostnameFilePath.nil?
+        return false unless @@HostnameFilePath =~ /\w/
+        return false unless File.exist?(@@HostnameFilePath)
+        return true
+      end
+
+      def is_dot_separated_string?(hnBuffer)
+        return true if /[^.]+\.[^.]+/ =~ hnBuffer
+        return false
+      end
+
+      def is_hostname_compliant?(hnBuffer)
+        # RFC 2181:
+        #   Size limit is 1 to 63 octets, so probably bytesize is appropriate method.
+        return false if hnBuffer.nil?
+        return false if /\./ =~ hnBuffer # Hostname by definition may not contain a dot.
+        return false if /:/ =~ hnBuffer # Hostname by definition may not contain a colon.
+        return false unless 1 <= hnBuffer.bytesize && hnBuffer.bytesize <= 63
+        return true
+      end
+
+      def is_like_ipv4_string?(hnBuffer)
+        return false unless /\A#{IPV4_Approximate_REGEX}\z/ =~ hnBuffer
+        qwa = hnBuffer.split('.')
+        return false unless qwa.length == 4
+        return false unless qwa[0].to_i != 0
+        i = 0
+        qwa.each do |quadwordstring|
+            bi = quadwordstring.to_i
+            # This may need more detail if 255 octets are sometimes allowed, but I don't think so.
+            return false unless 0 <= bi and bi < 255
+            i += 1
+        end
+        return true
+      end
+
+      def is_like_ipv6_string?(hnBuffer)
+        return true if /\A#{IPV6_REGEX}\z/ =~ hnBuffer
+        return false
+      end
+
+      def look_for_socket_class_host_address
+        hostname_buffer = nil
+
+        begin
+          hostname_buffer = Socket.gethostname
+        rescue => error
+          OMS::Log.error_once("Unable to get the Host Name using socket facility: #{error}")
+          return
+        end
+        hostname_buffer = clean_hostname_string(hostname_buffer)
+
+        if is_like_ipv4_string?(hostname_buffer) then
+          # If Socket.gethostname returns an IPv4 address,
+          # use the entire string.
+          @@Hostname = hostname_buffer
+        else
+          # Here we presume we got a usable hostname,
+          # though it could also be an IPV6, or other things,
+          # and more attention may be needed here later.
+          @@Hostname = clean_hostname_string(hostname_buffer)
```

Comment:
You already called `clean_hostname_string(hostname_buffer)` on line 545, so both the `if is_like_ipv4_string?(hostname_buffer)` case and the `else` case simply set `@@Hostname = hostname_buffer`. I believe this if/else can be replace with just the set line.